### PR TITLE
Add theming

### PR DESCRIPTION
**Pull request by amalv**
- Create theme provider
- Create color definitions

### CODE REVIEW

File: src/Theming/theme.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  ...STATUS_ENUM,
+} as const;
+
+export type ColorVariants = typeof COLORS[keyof typeof COLORS];
+
+export interface Theme {
+  color: Record<ColorVariants, string>;
+}
+
+export const theme: Theme = {
```
Comment (mikelgarciaurbina) [inline src/Theming/theme.tsx:35]: No estamos siguiendo la estructura de mui https://mui.com/material-ui/customization/default-theme/

**Review by mikelgarciaurbina — CHANGES_REQUESTED** [review]

File: src/Theming/colorDefinitions/blue.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  BLUE100: 'blue100',
+  BLUE200: 'blue200',
+  BLUE300: 'blue300',
+  BLUE400: 'blue400',
+  BLUE500: 'blue500',
+  BLUE600: 'blue600',
+  BLUE800: 'blue800',
+  BLUE900: 'blue900',
+
+  DARK_BLUE: 'dark_blue',
```
Comment (EduardoRG) [inline src/Theming/colorDefinitions/blue.ts:13]: According to DS2, these variants dark_* and light_* and older naming might be deprecated, am I right?

File: src/Theming/theme.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    ...GREEN_COLORS,
+    ...NEUTRAL_COLORS,
+    ...ORANGE_COLORS,
+    ...PINK_COLORS,
+    ...PURPLE_COLORS,
+    ...STATUS_COLORS,
+  },
+};
+
+export const LUIThemeProvider = ({
```
Comment (EduardoRG) [inline src/Theming/theme.tsx:47]: Do you remember @david-gomey if this Provider is useful right now? I thought it was related to having more than one instance of styled-components, but we fixed it by defining it as peer dependency.

**Review by EduardoRG — CHANGES_REQUESTED** [review]

File: src/Theming/theme.tsx (hunk 3 above)
Comment (david-gomey) [reply src/Theming/theme.tsx:47]: This Provider is only rendering children, `<>{children}</>`, if we want to create a component that automatically adds the theme perhaps should be `<ThemeProvider theme={theme}>{children}</ThemeProvider>` with ThemeProvider from styled-components.

Comment (david-gomey) [reply src/Theming/theme.tsx:47]: In my opinion, if we want to control from here the theme, perhaps it is useful. If we want to add more flexibility from the app that use the theme, perhaps we only need to expose the theme and the app should add `<ThemeProvider theme={theme}>{children}</ThemeProvider>`

Comment (EduardoRG) [reply src/Theming/theme.tsx:47]: Yes, I would remove it too if there's no specific use case for it.

**Review by mikelgarciaurbina — APPROVED** [review]

**Review by EduardoRG — APPROVED** [review]

**Review by david-gomey — APPROVED** [review]

**Pull request merged**